river/sync.go

```diff
@@ -479,3 +462,33 @@ func (r *River) doBulk(reqs []*elastic.BulkRequest) error {
 
 	return nil
 }
+
+// get mysql field value and convert it to specific value to es
+func (r *River) getFieldValue(col *schema.TableColumn, fieldType string, value interface{}) interface{} {
+	var fieldValue interface{}
+	switch fieldType {
+	case fieldTypeList:
+		v := r.makeReqColumnData(col, value)
+		if str, ok := v.(string); ok {
+			fieldValue = strings.Split(str, ",")
+		} else {
+			fieldValue = v
+		}
+
+	case fieldTypeDate:
+		if col.Type == schema.TYPE_NUMBER {
+			col.Type = schema.TYPE_DATETIME
+			switch value := value.(type) {
+			case int64:
```

Comment:
why we only care int64 and int32 here?

Reply:
oh, change it to
```
if col.Type == schema.TYPE_NUMBER {
			col.Type = schema.TYPE_DATETIME
			switch value := value.(type) {
			case int64:
				fieldValue = r.makeReqColumnData(col, time.Unix(value, 0).Format("2006-01-02 15:04:05"))
			case int:
				fieldValue = r.makeReqColumnData(col, time.Unix(int64(value), 0).Format("2006-01-02 15:04:05"))
			case int8:
				fieldValue = r.makeReqColumnData(col, time.Unix(int64(value), 0).Format("2006-01-02 15:04:05"))
			case int16:
				fieldValue = r.makeReqColumnData(col, time.Unix(int64(value), 0).Format("2006-01-02 15:04:05"))
			case int32:
				fieldValue = r.makeReqColumnData(col, time.Unix(int64(value), 0).Format("2006-01-02 15:04:05"))
			}
		}
```

Reply:
seem you can extract to a function like `toInt64(v) (int64, bool)` so we can simplify the switch case.